src/modules/layout/components/Header.js

```diff
@@ -112,6 +111,21 @@ class Header extends React.PureComponent<PropsType> {
     })
   }
 
+  showLanguageChange (): boolean {
+    const {routeKey, routeMapping} = this.props
+    const route = routeMapping[routeKey]
+
+    if (!route) {
+      return false
+    }
+
+    return !route.peek
+  }
+
+  showSearch (): boolean {
+    return this.showLanguageChange()
```

Comment:
maybe extract a method `isPeekScreen` from `showLanguageChange` and call that for better readability